src/python/zquantum/core/estimator.py

```diff
@@ -22,10 +25,25 @@ def get_estimated_expectation_values(
             backend (QuantumBackend): the backend that will be used to run the circuit
             circuit (Circuit): the circuit that prepares the state.
             target_operator (List[SymbolicOperator]): List of target functions to be estimated.
+            n_samples (int): Number of measurements done on the unknown quantum state. 
+            epsilon (float): an error term.
+            delta (float): a confidence term.
 
         Returns:
             ExpectationValues: expectation values for each term in the target operator.
         """
+        estimator_name = type(self).__name__
+        self._ignore_parameter(estimator_name, "epsilon", epsilon)
+        self._ignore_parameter(estimator_name, "delta", delta)
+
+        if n_samples is not None:
+            save = backend.n_samples
+            backend.n_samples = n_samples
+            expectation_values = backend.get_expectation_values(
```

Comment:
This whole `if` is confusing to me – why do you calculate expectation values if you don't use them at all?

Reply:
the return statement should be returning `expectation_values`. just a bug, will fix in the next commit.

Reply:
Typo in `arugment` ;)